### PR TITLE
Have ERT handle UserWarning from ert_script

### PR DESCRIPTION
**Pull request by andreas-el**
Relates to https://github.com/equinor/semeio/pull/611
Resolves https://github.com/equinor/semeio/issues/584

**Issue**
Resolves #my_issue

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: tests/unit_tests/job_queue/test_ert_script.py [hunk 1]
```diff
@@ -22,12 +22,24 @@ def run(self, arg):
...
+        raise UserWarning("Custom user warning")
+
+
 def test_ert_script_return_ert():
     script = ReturnErtScript("ert", storage=None)
     result = script.initializeAndRun([], [])
     assert result == "ert"
 
 
+def test_failing_ert_script_provide_user_warning():
```
Comment (berland) [inline tests/unit_tests/job_queue/test_ert_script.py:36]: provide -> provides

File: src/ert/config/ert_script.py [hunk 2]
```diff
@@ -107,6 +107,9 @@ def initializeAndRun(
             error_msg = "Script cancelled (CTRL+C)"
             self.output_stack_trace(error=error_msg)
             return None
+        except UserWarning as uw:
```
Comment (berland) [inline src/ert/config/ert_script.py:110]: Does it make sense to treat the Warning as an Exception?

Comment (andreas-el) [reply src/ert/config/ert_script.py:110]: But `UserWarning` is technically an exception? I think I need to catch this to be able to redirect the information to the dialog?

Comment (berland) [reply src/ert/config/ert_script.py:110]: It is strange, but probably just how Python is.

**Review by berland — APPROVED** [review]
LGTM

**Pull request merged**